sessionctx/variable/sysvar.go

```diff
@@ -1570,6 +1570,20 @@ var defaultSysVars = []*SysVar{
 			return nil
 		},
 	},
+	{Scope: ScopeGlobal, Name: TiDBAuditLogVersion, Hidden: true, Value: strconv.Itoa(2), Type: TypeUnsigned, MinValue: 1, MaxValue: 2,
+		GetGlobal: func(s *SessionVars) (string, error) {
+			return strconv.FormatInt(AuditLogVersion.Load(), 10), nil
+		},
+		SetGlobal: func(s *SessionVars, sVal string) error {
+			val, err := strconv.ParseInt(sVal, 10, 64)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			AuditLogVersion.Store(val)
+			return nil
+		},
+	},
```

Comment:
I think you should be able to do the filtering from within the plugin code based on the event? This will be more consistent because then you have one Audit API, and 2 versions of "the audit log plugin".

For the variable name, I think you can make it more descriptive: `Name: TiDBAuditLogCaptureEvents, Type: TypeEnum, PossibleValues: []string{"BEFORE", "AFTER", "BEFOREANDAFTER"}`.

We are also trying to stop using hidden variables, but if you make this part of the plugin code it will only be present for those that load the plugin, which is great.

Reply:
Fine. I'll consider about it.